app/furniture/marketplace/order.rb

```diff
@@ -52,5 +52,135 @@ def marketplace_name
     def price_total
       [product_total, delivery_fee, tax_total].compact.sum
     end
+
+    def square_client
+      @square_client ||= Square::Client.new(access_token: marketplace.square_access_token, environment: marketplace.square_environment)
+    end
+
+    def send_to_square_seller_dashboard
+      square_create_order_response = create_square_order
+      square_create_payment_response = create_square_order_payment(square_create_order_response.body.order[:id], square_create_order_response.body.order[:total_money][:amount])
+
+      # This data is intended for use in debugging, etc... until we further
+      # the Square integration productize
+      {
+        order_id: square_create_order_response.body.order[:id],
+        payment_id: square_create_payment_response.body.payment[:id]
+      }
+    end
+
+    private def create_square_order
+      square_create_order_body = build_square_create_order_body(marketplace)
+      square_client.orders.create_order(body: square_create_order_body)
+    end
+
+    # NOTE: Square requires that orders are paid in order to show up in the Seller
+    # Dashboard
+    private def create_square_order_payment(square_order_id, square_order_total_price)
+      square_location_id = marketplace.settings["square_location_id"]
+      space_id = marketplace.space.id
+      square_create_payment_body = build_square_create_order_payment_body(
+        square_order_id,
+        square_location_id,
+        space_id,
+        square_order_total_price
+      )
+
+      @square_client.payments.create_payment(body: square_create_payment_body)
+    end
+
+    # NOTE: Square requires that orders include fulfillments in order to show up
+    # in the Seller Dashboard
+    # See: https://developer.squareup.com/docs/orders-api/create-orders
+    private def build_square_create_order_body(marketplace)
+      idempotency_key = "#{id}_#{8.times.map { rand(10) }.join}"
+      location_id = marketplace.settings["square_location_id"]
+      customer_id = shopper.id
+
+      stripe_fee_per_line_item = (payment_processor_fee_cents / ordered_products.sum(&:quantity))
```

Comment:
@zspencer @anaulin I can't tell if this is tacky or not, but it quickly felt like the best way to make sure line items are separated in the Square order and the _vendor share_ total is accurately calculated.

Reply:
This calculation feels like a dangerous idea from a book-keeping perspective. They are selling the product for $10, not $10 - a_fraction_of_the_stripe_fees.

Reply:
Yeah I figured as much. There's gotta be a better way around this. Square seems to compute and validate the line item total generated during order creation against the payment total during payment creation. If they don't match, the payment fails.

The trick is to have the vendor/seller line items appear with the correct amounts. Need to think on this more. It could be we add a negative service charge with the order creation data for the Stripe fee. But I'm not sure sellers are supposed to be aware of the Stripe fee?

Reply:
Can we tell Square that the Payment is for the total?